src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-snapshot-list/rbd-snapshot-list.component.ts

```diff
@@ -140,12 +141,19 @@ export class RbdSnapshotListComponent implements OnInit, OnChanges {
   }
 
   private openSnapshotModal(taskName: string, oldSnapshotName: string = null) {
+    let snapName;
     this.modalRef = this.modalService.show(RbdSnapshotFormComponent);
     this.modalRef.content.poolName = this.poolName;
     this.modalRef.content.imageName = this.rbdName;
     if (oldSnapshotName) {
-      this.modalRef.content.setSnapName(this.selection.first().name);
+      snapName = this.selection.first().name;
+      this.modalRef.content.setEditing();
+    } else {
+      // Auto-create a name for the snapshot: <image_name>_<timestamp_ISO_8601>
+      // https://en.wikipedia.org/wiki/ISO_8601
+      snapName = `${this.rbdName}-${moment().utc().format('YYYYMMDD[T]HHmmss[Z]')}`;
     }
+    this.modalRef.content.setSnapName(snapName);
```

Comment:
Could you please add two tests to verify the behavior? I suggest for the first test you call `openCreateSnapshotModal` and verify the automatic name creation, for the other you call `openEditSnapshotModal` and verify the old behavior.